actioncable/lib/action_cable/subscription_adapter/redis.rb

```diff
@@ -0,0 +1,46 @@
+begin
+  require 'em-hiredis'
+  require 'redis'
+rescue Gem::LoadError => e
+  raise Gem::LoadError, "You are trying to use the Redis ActionCable adapter, but do not have the proper gems installed. Add `gem 'em-hiredis'` and `gem 'redis'` to your Gemfile (and ensure its version is at the minimum required by ActionCable)."
+end
+
+module ActionCable
+  module SubscriptionAdapter
+    class Redis < Base
+      # The redis instance used for broadcasting. Not intended for direct user use.
+      def broadcast(channel, payload)
+        redis_conn.publish(channel, payload)
+      end
+
+      def subscribe(channel, callback, success_callback = nil)
+        hi_redis_conn.pubsub.subscribe(channel, &callback).tap do |result|
+          result.callback(&success_callback) if success_callback
+        end
+      end
+
+      def unsubscribe(channel, callback, success_callback = nil)
+        if hi_redis_conn.pubsub.unsubscribe_proc(channel, callback)
+          success_callback.call if success_callback
+        end
+      end
+
+      private
+
+      # The EventMachine Redis instance used by the pubsub adapter.
+      def hi_redis_conn
```

Comment:
I'd go with `subscription_redis_connection`. We generally avoid abbreviations like `conn` (or we certainly should) and "hi/em" is a implementational detail.